topologytest/hlv_test.go

```diff
@@ -72,9 +135,47 @@ func TestHLVCreateDocumentSingleActor(t *testing.T) {
 		t.Run(tc.description(), func(t *testing.T) {
 			peers, _ := setupTests(t, tc.topology, tc.activePeerID)
 
+			docID := tc.docID("")
 			docBody := []byte(fmt.Sprintf(`{"peer": "%s", "topology": "%s"}`, tc.activePeerID, tc.description()))
-			docVersion := peers[tc.activePeerID].CreateDocument(tc.collectionName(), tc.docID(), docBody)
-			waitForVersionAndBody(t, tc, peers, docVersion, docBody)
+			docVersion := peers[tc.activePeerID].CreateDocument(tc.collectionName(), tc.docID(""), docBody)
+			bodyVersion := BodyAndVersion{
+				docMeta:    docVersion,
+				body:       docBody,
+				updatePeer: tc.activePeerID,
+			}
+			waitForVersionAndBody(t, tc, peers, docID, bodyVersion)
+		})
+	}
+}
+
+// TestHLVCreateDocumentMultiActorConflict:
+//   - Create conflicting docs on each peer
+//   - Wait for docs last write to be replicated to all other peers
+func TestHLVCreateDocumentMultiActorConflict(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyCRUD, base.KeyImport, base.KeyVV)
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Panics against rosmar, CBG-4378")
+	} else {
+		t.Skip("Flakey failures on multi actor conflicting writes, CBG-4379")
+	}
+	for _, tc := range getMultiActorTestCases() {
+		if strings.Contains(tc.description(), "CBL") {
+			// Test case flakes given the WaitForDocVersion function only waits for a docID on the cbl peer. We need to be
+			// able to wait for a specific version to arrive over pull replication
+			t.Skip("We need to be able to wait for a specific version to arrive over pull replication, CBG-4257")
+		}
+		t.Run(tc.description(), func(t *testing.T) {
+			peers, replications := setupTests(t, tc.topology, "")
+
+			stopPeerReplications(t, replications)
+
+			docID := tc.docID("")
```

Comment:
What do you think about having something like `singleDocID()` with no argument and `peerDocID(peerName string)` for readability?